cypress/integration/api-class_spec.js

```diff
@@ -151,6 +154,43 @@ describe("Api class", () => {
 		});
 	});
 
+	describe("setCustomHeaders()", () => {
+		it("should change the customHeaders", () => {
+			const newCustomHeaders = {
+				"x-custom-1": "1",
+				"x-custom-2": "2",
+			};
+			config.api.setCustomHeaders(newCustomHeaders);
+
+			expect(config.api.customHeaders).to.be.equal(newCustomHeaders);
+		});
+		it("should throw an error when using something other than an Object as customHeaders", () => {
+			filterPrimitives([
+				"Object", "undefined",
+			]).forEach((set) => {
+				expect(() => config.api.setCustomHeaders(set.value))
+					.to.throw(`Expected \`customHeaders\` to be of type \`object\` but received type \`${set.type}\``);
+			});
+		});
+
+		it("should throw an error when using a blacklisted header as one of the custom headers", () => {
+			const blacklistedHeaderKey = CUSTOMHEADER_BLACKLIST[
+				Math.floor(Math.random() * CUSTOMHEADER_BLACKLIST.length)
```

Comment:
Is het logisch om random 1 header te checken? Zou je niet gewoon altijd alle headers checken

Als je het random doet zou je miss code kunnen schrijven die 1 header toestaat maar de tests dan daarop niet afgaat. En dan later bij een niet gerelateerde change opeens wel fout gaat. Of zie ik dat verkeerd?

Reply:
> Of zie ik dat verkeerd?

Nee je hebt gelijk. Ik dacht dat snelheid een issue zou zijn dus daarom testte ik maar 1 header, maar dit kan makkelijk met alle headers.